rosidl_typesupport_opendds_cpp/resource/srv__type_support.cpp.em

```diff
@@ -178,6 +179,7 @@ void * create_requester__@(service.namespaced_type.name)(
     new (requester) RequesterType(requester_params);
   } catch (...) {
     RMW_SET_ERROR_MSG("C++ exception during construction of Requester");
+    free(requester);
```

Comment:
I know we've been having the conversation about handling leaks in standup but is there any reason why these can't be smart pointers either the `std` or `OpenDDS` variety?

Reply:
With the current API, we are not ready for this.  The function here knows about `allocator` but not `deallocator`.  It needs to be handled as a separate task.